cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/SparkCqlEvaluator.java

```diff
@@ -98,16 +114,24 @@
 
     @DynamicParameter(names = { "-l",
             "--library" }, description = "One or more library=version key-value pair(s), as defined in the jobs file, that describe the libraries that should be run in this evaluation. Defaults to all libraries. Specify multiple libraries by providing a separate option for each library.", required = false)
-    Map<String, String> libraries = new HashMap<>();
+    public Map<String, String> libraries = new HashMap<>();
 
     @Parameter(names = { "-e",
             "--expression" }, description = "One or more expression names, as defined in the context-definitions file, that should be run in this evaluation. Defaults to all expressions.", required = false)
-    Set<String> expressions = new HashSet<>();
+    public Set<String> expressions = new HashSet<>();
+
+    @Parameter(names = { "-n",
+            "--output-partitions" }, description = "Number of partitions to use when storing data", required = false)
+    public Integer outputPartitions = null;
 
     @Parameter(names = { "--debug" }, description = "Enables CQL debug logging")
     public boolean debug = false;
 
-    public SparkTypeConverter typeConverter;
+    protected SparkTypeConverter typeConverter;
+
+    protected static ThreadLocal<CqlEvaluationRequests> jobSpecification = new ThreadLocal<>();
+    protected static ThreadLocal<CqlLibraryProvider> libraryProvider = new ThreadLocal<>();
+    protected static ThreadLocal<CqlTerminologyProvider> terminologyProvider = new ThreadLocal<>();
```

Comment:
Pedantic: I'm not sure what to say, but I feel like some comment stating that these providers need to be unique per thread would be helpful for future devs.  Making any change to their scope will conflict with how `CQLContextFactory` is currently designed (namely, the `CONTEXT_CACHE` map).